### PR TITLE
Fix version check workflow. Breaks on PR from fork.

### PR DESCRIPTION
**Pull request by jacobcrawford**
The checkout tried to look in the original repo and not the forked one.
We now parse in the reference from the pull request, which points the where the changes happened.

This will make changes from a fork work as well.

### CODE REVIEW

File: .github/workflows/check_helm_version.yaml [hunk 1]
```diff
@@ -26,7 +26,9 @@ jobs:
 
       - uses: actions/checkout@v2
         with:
-          ref: ${{ github.head_ref }}
+          fetch-depth: 0
```
Comment (klausenbusk) [inline .github/workflows/check_helm_version.yaml:29]: Why do we need the whole history?

File: yggdrasil/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: yggdrasil
 description: A Helm chart for deploying services and applications
 type: application
-version: v1.0.4
+version: v1.0.5
```
Comment (jaxels10) [inline yggdrasil/Chart.yaml:5]: We need to bump both the yggdrasil and nidhogg chart versions so they follow each other. But wait, my current PR also bumps to 1.0.5.... race condition for the merge

Comment (jaxels10) [reply yggdrasil/Chart.yaml:5]: @jacobcrawford I guess I won the battle for now, we need to bump both this and nidhogg up to v1.0.6 in this PR

File: .github/workflows/check_helm_version.yaml (hunk 1 above)
Comment (jacobcrawford) [reply .github/workflows/check_helm_version.yaml:29]: If we did not need it before I think we do not need It now :)

File: yggdrasil/Chart.yaml (hunk 2 above)
Comment (jacobcrawford) [reply yggdrasil/Chart.yaml]: gg

**Review by jaxels10 — APPROVED** [review]
LGTM

**Pull request merged**